core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -936,30 +931,31 @@ private object RTS {
     }
 
     @tailrec
-    private final def kill0[E2](t: Throwable, k: Callback[E, Unit]): Async[E2, Unit] = {
+    private final def kill0[E2](ts: List[Throwable], k: Callback[E, Unit]): Async[E2, Unit] = {
       killed = true
 
       val oldStatus = status.get
 
       oldStatus match {
-        case Executing(t0, joiners, killers) =>
-          if (!status.compareAndSet(oldStatus, Executing(t0.orElse(Some(t)), joiners, k :: killers))) kill0(t, k)
+        case Executing(ts0, joiners, killers) =>
+          if (!status.compareAndSet(oldStatus, Executing(Some(ts0.getOrElse(Nil) ++ ts), joiners, k :: killers)))
```

Comment:
Note the ordering matters, though: if we want the first errors that killed it to be at the end of the list, then this would be correct.